smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -879,36 +877,41 @@ private void writeRequestQueryParam(
         String queryValue = getInputValue(
             context,
             binding.getLocation(),
-            "input." + memberName + memberAssertionComponent,
+            "input[" + context.getStringStore().var(memberName) + "]" + memberAssertionComponent,
             binding.getMember(),
             target
         );
 
         writer.addImport("expectNonNull", "__expectNonNull", TypeScriptDependency.AWS_SMITHY_CLIENT);
 
-        if (Objects.equals("input." + memberName + memberAssertionComponent, queryValue)) {
+        if (
```

Comment:
nit - this if statement is getting kind of hard to read. Maybe break this up by storing the predicate string as a variable first.